### DISCUSSION TITLE
What's in the generated Private.SourceBuilt.Prebuilts tarball?

### DISCUSSION THREAD

**Discussion by tmds** [dotnet/source-build#3595] [Q&A] [answered]
What files get stored in `Private.SourceBuilt.Prebuilts`?

For example: when building dotnet/dotnet release/8.0.1xx then the tarball contains these 4 files:
```
FSharp.Compiler.Service.43.8.100.nupkg
FSharp.Compiler.Service.43.8.100-preview.23421.4.nupkg
FSharp.Core.8.0.100-beta.23421.4.nupkg
FSharp.Core.8.0.100.nupkg
```

Files with the exact same name are also present in the `Private.SourceBuilt.Artifacts` that is produced by the same build.
I think that means these files were produced by the build?
If so: why are these still added to the `Prebuilts` tarball? Or if they are meant to be there: why are there no other packages in `Private.SourceBuilt.Prebuilts` that were prebuilts to the build?

cc @MichaelSimons @mthalman @omajid

**Comment by mthalman** [comment]
This was a long known issue that was just recently fixed: https://github.com/dotnet/source-build/issues/2989

**Reply by tmds** [reply]
I see, the presence of the FSharp packages is misguiding then.

What is the Private.SourceBuilt.Prebuilts tarball supposed to contain?
Can the same filename ever appear in Private.SourceBuilt.Artifacts that was produced along side, or would that be a bug?

**Reply by mthalman** [reply]
The Prebuilts tarball should only includes those packages that were detected as prebuilts. I suppose it might be possible for such a package to also exist in the Artifacts tarball but that would indicate the presence of a poison leak which we do have checks for.

**Reply by tmds** [reply, marked as answer]
> The Prebuilts tarball should only includes those packages that were detected as prebuilts.

And a prebuilt is a package that gets consumed by the build, but that isn't produced?
Is that a proper definition?

**Reply by mthalman** [reply]
Yes, exactly

**Reply by tmds** [reply]
Thanks for making it more clear!
